### PR TITLE
deps: more permissive mst peerDep, update mst devDep

### LINKED ISSUE #32: mst peerDep is outdated
When running `npm i` this error occurs:
```
PS C:\Users\alexa\vermehrung> npm i
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: vermehrung.ch@1.14.10
npm ERR! Found: mobx-state-tree@5.1.8
npm ERR! node_modules/mobx-state-tree
npm ERR!   mobx-state-tree@"5.1.8" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer mobx-state-tree@"^3.2.1" from mst-persist@0.1.3
npm ERR! node_modules/mst-persist
npm ERR!   mst-persist@"0.1.3" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! C:\Users\alexa\AppData\Local\npm-cache\_logs\2023-07-12T13_22_49_699Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\alexa\AppData\Local\npm-cache\_logs\2023-07-12T13_22_49_699Z-debug-0.log
```

This forces to install using `npm i --force` which is not recommended.

As no one will be using mobx-state-tree@^3.2.1 any more, it would help to update the peer dependency to a newer version to `Fix the upstream dependency conflict`.

### PR DESCRIPTION
**Pull request by barbalex**
resolves https://github.com/agilgur5/mst-persist/issues/32

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
     "changelog": "changelog-maker"
   },
   "peerDependencies": {
-    "mobx-state-tree": "^3.2.1"
+    "mobx-state-tree": "^5.1.8"
```
Comment (agilgur5) [inline package.json]: So this change would be breaking, instead we can just make the existing range more permissive to allow for any version above:

```suggestion
    "mobx-state-tree": ">=3.2.1"
```

This change would be a patch bump instead

**Review by agilgur5 — CHANGES_REQUESTED** [review]
You were able to test and confirm that this was working on MST 5, right? Seems like there weren't too many changes in [4.0.0](https://github.com/mobxjs/mobx-state-tree/blob/v5.1.8/changelog.md#400) or [5.0.0](https://github.com/mobxjs/mobx-state-tree/releases/tag/v5.0.0) thankfully

The CI on this repo is still on Travis, so I need to migrate that to GH Actions to get it to run on each PR

**Review by agilgur5 — COMMENTED** [review]
We should update the devDeps as well to make sure MST 5 is tested against. That'll require an update of the MobX devDep as well

**Comment by agilgur5** [comment]
> The CI on this repo is still on Travis, so I need to migrate that to GH Actions to get it to run on each PR

CI has been migrated as of #34 now. If you rebase off `main`, that should get it to run on your PR

**Comment by barbalex** [comment]
> You were able to test and confirm that this was working on MST 5, right?

Yes, in two of my projects it has been running flawlessly for years

**Comment by barbalex** [comment]
Not sure if I did all that's needed. Please check

**Review by agilgur5 — APPROVED** [review]
updated devDeps myself, LGTM now! all tests still passing 🙂 

also had to update some one-line CI settings

**Comment by agilgur5** [comment]
Released in [`0.1.4`](https://github.com/agilgur5/mst-persist/releases/tag/v0.1.4).

Thanks for your continued contribution @barbalex!

**Pull request merged**